src/drt/src/dr/FlexDR_init.cpp

```diff
@@ -35,6 +35,20 @@ using namespace std;
 using namespace fr;
 namespace bgi = boost::geometry::index;
 
+bool FlexDRWorker::isRoutePatchWire(const frPatchWire* pwire) const
+{
+  const auto& gridBBox = getRouteBox();
+  Point origin = pwire->getOrigin();
+  return isInitDR() ? gridBBox.overlaps(origin) : gridBBox.intersects(origin);
```

Comment:
This is quite subtle - what's the logic behind this?

Reply:
Not sure yet. This is the previous implementation of the shared border model and how it treats items specially in the first iteration. My guess would be that the boundary pins in the first iteration are based on the track assignment results not boundary objects. Thus we don't ripup the border shapes in the first iteration because we wouldn't reroute them and we would leave the unpatched object as it is which would be violating.